plugin/builtin/patchtransformer/PatchTransformer.go

```diff
@@ -13,101 +13,120 @@ import (
-			p.loadedPatch.AllowNameChange()
-		}
-		if p.Options["allowKindChange"] {
-			p.loadedPatch.AllowKindChange()
+		p.smPatches = patchesSM
+		for _, loadedPatch := range p.smPatches {
+			if p.Options["allowNameChange"] {
+				loadedPatch.AllowNameChange()
+			}
+			if p.Options["allowKindChange"] {
+				loadedPatch.AllowKindChange()
+			}
 		}
 	} else {
-		p.decodedPatch = patchJson
+		p.jsonPatches = patchesJson
 	}
 	return nil
 }
 
 func (p *plugin) Transform(m resmap.ResMap) error {
-	if p.loadedPatch == nil {
-		return p.transformJson6902(m, p.decodedPatch)
+	if p.smPatches != nil {
+		return p.transformStrategicMerge(m)
 	}
-	// The patch was a strategic merge patch
-	return p.transformStrategicMerge(m, p.loadedPatch)
+	return p.transformJson6902(m)
 }
 
-// transformStrategicMerge applies the provided strategic merge patch
-// to all the resources in the ResMap that match either the Target or
-// the identifier of the patch.
-func (p *plugin) transformStrategicMerge(m resmap.ResMap, patch *resource.Resource) error {
-	if p.Target == nil {
-		target, err := m.GetById(patch.OrgId())
+// transformStrategicMerge applies each loaded strategic merge patch
+// to the resource in the ResMap that matches the identifier of the patch.
+// If only one patch is specified, the Target can be used instead.
+func (p *plugin) transformStrategicMerge(m resmap.ResMap) error {
+	if p.Target != nil {
+		if len(p.smPatches) > 1 {
+			// detail: https://github.com/kubernetes-sigs/kustomize/issues/5049#issuecomment-1440604403
+			return fmt.Errorf("Multiple Strategic-Merge Patches in one `patches` entry is not allowed to set `patches.target` field: %s", p.patchSource)
+		}
+
+		// single patch
+		patch := p.smPatches[0]
+		selected, err := m.Select(*p.Target)
 		if err != nil {
-			return err
+			return fmt.Errorf("unable to find patch target %q in `resources`: %w", p.Target, err)
 		}
-		return target.ApplySmPatch(patch)
+		return errors.Wrap(m.ApplySmPatch(resource.MakeIdSet(selected), patch))
 	}
-	selected, err := m.Select(*p.Target)
-	if err != nil {
-		return err
+
+	for _, patch := range p.smPatches {
+		target, err := m.GetById(patch.OrgId())
+		if err != nil {
+			return fmt.Errorf("unable to find patch target %q: %w", patch.OrgId().Name, err)
+		}
+		if err := target.ApplySmPatch(patch); err != nil {
+			return errors.Wrap(err)
+		}
 	}
-	return m.ApplySmPatch(resource.MakeIdSet(selected), patch)
+	return nil
 }
 
-// transformJson6902 applies the provided json6902 patch
-// to all the resources in the ResMap that match the Target.
-func (p *plugin) transformJson6902(m resmap.ResMap, patch jsonpatch.Patch) error {
+// transformJson6902 applies json6902 Patch to all the resources in the ResMap that match Target.
+func (p *plugin) transformJson6902(m resmap.ResMap) error {
 	if p.Target == nil {
-		return fmt.Errorf("must specify a target for patch %s", p.Patch)
+		return fmt.Errorf("must specify a target for patch %s", p.patchText)
```

Comment:
nit: Would it be helpful if we mentioned that this is a json patch? We might confuse users who think the target is optional for strategic merge patches.

I also considered printing `p.patchSource` instead of `p.patchText` to maybe enable the user to find the problematic patch more quickly, but I don't feel strongly. The error string might look something like

`must specify a target for JSON patch %s`

Reply:
> I also considered printing p.patchSource instead of p.patchText

That is right!
I'll fix this message.